src/app/games/tafl/TaflRules.ts

```diff
@@ -0,0 +1,462 @@
+         *      6: if king not capturable by empty throne               -> no capture
+         *      7: if king capturable by 2                              -> capture king (1 invader + throne)
+         *      8: else if two-other-coord are invader                  -> capture king (3 invaders + throne)
+         *  9: allied is an invader
+         *     10: if king is capturable by two                         -> capture king (2 invaders)
+         *     11: if 2 others around king are invaders                 -> capture king (4 invaders)
+         * So these are the different victory way for the invaders :
+         * - 2 invaders
+         * - 1 invaders 1 empty-throne
+         * - 3 invaders 1 throne
+         * - 2 invaders 1 throne 1 border
+         * - 3 invaders 1 border
+         * - 4 invaders
+         */
+        const kingCoord: Coord = landingPiece.getNext(d);
+
+        const {
+            backCoord, back, backInRange,
+            leftCoord, left,
+            rightCoord, right,
+        } = this.getSurroundings(kingCoord, d, player, state.board);
+
+        if (backInRange === false) { // /////////////////////////////////////////////////////////////////// 1
+            return this.captureKingAgainstTheWall(left, leftCoord, right, rightCoord, kingCoord, state); // 2, 3
+        }
+        if (back === RelativePlayer.NONE && this.isThrone(state, backCoord)) { // ///////////////////////// 4
+            return this.captureKingAgainstThrone(backCoord, kingCoord, left, right);
+        }
+        if (back === RelativePlayer.PLAYER) {
+            return this.captureKingWithAtLeastASandwhich(state, kingCoord, left, leftCoord, right, rightCoord);
+        }
+        return null;
+    }
+    public getSurroundings(c: Coord, d: Orthogonal, player: Player, board: Table<TaflPawn>)
+    : { backCoord: Coord, back: RelativePlayer, backInRange: boolean,
+        leftCoord: Coord, left: RelativePlayer,
+        rightCoord: Coord, right: RelativePlayer }
+    {
+        const backCoord: Coord = c.getNext(d); // the piece that just move came from the front direction (by definition)
+        const backInRange: boolean = backCoord.isInRange(this.config.WIDTH, this.config.WIDTH);
+        const back: RelativePlayer = backInRange ?
+            this.getRelativeOwner(player, backCoord, board) :
+            RelativePlayer.NONE;
+
+        const leftCoord: Coord = c.getLeft(d);
+        const leftInRange: boolean = leftCoord.isInRange(this.config.WIDTH, this.config.WIDTH);
+        const left: RelativePlayer = leftInRange ?
+            this.getRelativeOwner(player, leftCoord, board) :
+            RelativePlayer.NONE;
+
+        const rightCoord: Coord = c.getRight(d);
+        const rightInRange: boolean = rightCoord.isInRange(this.config.WIDTH, this.config.WIDTH);
+        const right: RelativePlayer = rightInRange ?
+            this.getRelativeOwner(player, rightCoord, board) :
+            RelativePlayer.NONE;
```

Comment:
y'a pas moyen de raccourcir tout ça avec un for dir in [right, left, front, back] ?

Reply:
Un petit sub ticket "enhance getSurrounding" in Tafl :D ?

Reply:
ya, voo pouvez ticketer !